File: app/channels/comments_channel.rb
class CommentsChannel < ApplicationCable::Channel
  def subscribe(data)
    commentable = data['commentable_type'].classify.constantize.find(data['commentable_id'])
    raise RecordNotFound unless commentable

    stream_for "#{data['commentable_type']}/#{commentable.id}/comments"
  end
end

File: app/javascript/channels/comments_channel.js
import consumer from "./consumer"

consumer.subscriptions.create("CommentsChannel", {
  connected() {
    let question = $('.question-data')

    if (!question) {
      return;
    }
    console.log('Connection result ', this.perform('subscribe', {
      commentable_id: question.data('question-id'),
      commentable_type: 'Question'
    }))
  },

  received(data) {
    if (gon.user_id === data.user_id) {
      return;
    }

    $(`#${data.commentable_type.toLowerCase()}-${data.commentable_id} .comments-list`).append(data.body);
  }
});
